pkg/session/serve.go

```diff
@@ -264,19 +269,45 @@ func (s *Session) serve() {
 
 		case "triggerComponentCheck":
 			if payload.ComponentName != "" {
-				comp := s.componentsRegistry.Get(payload.ComponentName)
-				if comp == nil {
-					log.Logger.Warnw("component not found", "name", payload.ComponentName)
-					response.ErrorCode = http.StatusNotFound
-					break
-				}
+				switch payload.TagName {
+				case "":
+					comp := s.componentsRegistry.Get(payload.ComponentName)
+					if comp == nil {
+						log.Logger.Warnw("component not found", "name", payload.ComponentName)
+						response.ErrorCode = http.StatusNotFound
+						break
+					}
 
-				rs := comp.Check()
-				response.States = apiv1.GPUdComponentHealthStates{
-					{
-						Component: payload.ComponentName,
-						States:    rs.HealthStates(),
-					},
+					rs := comp.Check()
+					response.States = apiv1.GPUdComponentHealthStates{
+						{
+							Component: payload.ComponentName,
+							States:    rs.HealthStates(),
+						},
+					}
```

Comment:
we should merge these lines with the default branch..
1. get components by component name or tags
2. for each component, run check and append the result.. 
3. return